public/app/features/transformers/extractFields/extractFields.ts

```diff
@@ -86,7 +86,7 @@ export function addExtractedFields(frame: DataFrame, options: ExtractFieldsOptio
     for (const [key, val] of Object.entries(obj)) {
       let buffer = values.get(key);
       if (buffer == null) {
-        buffer = new Array(count);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Indexed_collections#sparse_arrays

So it looks like this sparse array is causing this bug, but making this array dense will eat up a lot more memory, especially for sparse fields.